src/backend/metal/CommandBufferMTL.mm

```diff
@@ -29,111 +29,182 @@
 namespace backend { namespace metal {
 
     namespace {
-        struct CurrentEncoders {
-            Device* device;
 
+        struct GlobalEncoders {
             id<MTLBlitCommandEncoder> blit = nil;
-            id<MTLComputeCommandEncoder> compute = nil;
-            id<MTLRenderCommandEncoder> render = nil;
 
-            void EnsureNoBlitEncoder() {
-                ASSERT(render == nil);
-                ASSERT(compute == nil);
+            void Finish() {
                 if (blit != nil) {
                     [blit endEncoding];
                     blit = nil;  // This will be autoreleased.
                 }
             }
 
             void EnsureBlit(id<MTLCommandBuffer> commandBuffer) {
-                ASSERT(render == nil);
-                ASSERT(compute == nil);
                 if (blit == nil) {
+                    Finish();
```

Comment:
Isn't this a no-op because (blit == nil) and (blit != nil)?

Reply:
Done. I'll add it back when we have multiple command encoders at the top-level (if we need them sometime)